### PR TITLE
Metrics refactor

### PR DESCRIPTION
**Pull request by efkag**
Added code for the Sig2Noise ration and Sig2 Integral ratio.
The Gaussian method required the implementation of an MSe loss

### CODE REVIEW

**Comment by efkag** [comment]
@amanyazevedoamin We should add the __pycache__ folder to the git ignore probably

**Comment by efkag** [comment]
> @amanyazevedoamin We should add the **pycache** folder to the git ignore probably

Nevermind I just saw that you have added onto the gitignore that is on master but I branched out earlier

File: QualityMetrics.py [hunk 1]
```diff
@@ -6,6 +6,7 @@
 from scipy.signal import butter, filtfilt, lfilter
 
 device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+print("device", device)
 
 
 # Specify the Butterworth Filter
```
Comment (amanyazevedoamin) [inline QualityMetrics.py:12]: Happy for you to delete the old SNR stuff

File: examples/snr-example.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+path = os.path.join(os.path.dirname(__file__), os.pardir)
+fwd = os.path.dirname(__file__)
+sys.path.append(path)
+
+from QualityMetrics import Sig2Noise, Sig2Integ
+import torch
+
+some_rf = torch.rand(360)
+
+snr = Sig2Noise().Fit()
```
Comment (amanyazevedoamin) [inline examples/snr-example.py:12]: Sig2Noise doesn't need the .Fit() called right? Shouldn't it be like the example below, with just setting up the object and calling .Eval

File: QualityMetrics.py [hunk 3]
```diff
@@ -37,3 +38,103 @@ def SNR(rf, filter=False):
 
 # For find peaks function, something like..
 #peaks, _ = scipy.signal.find_peaks(r, height=0.9 * 24)
+
+#TODO: The frist 2 metric may need to implemented such that we process multiple RFs
```
Comment (amanyazevedoamin) [inline QualityMetrics.py:42]: At the point of processing RRFFs, I stopped doing parallel operations because things were so fast anyway. E..g on my machine it takes ~ 1 second to get the SNR for 100 RRFFs through list comp. So maybe something to consider in future if we are going thousands and thousands of RRFFs? But certainly not something to worry about at the present moment.

**Review by amanyazevedoamin — CHANGES_REQUESTED** [review]
Thanks for these! Please see the comments, mainly about removing the old SNR code - then I'm happy to merge.

**Review by amanyazevedoamin — APPROVED** [review]

File: QualityMetrics.py (hunk 1 above)
Comment (efkag) [reply QualityMetrics.py]: Done

File: QualityMetrics.py (hunk 3 above)
Comment (efkag) [reply QualityMetrics.py]: We can indeed easily refactor this coed in the future to handle multiple RFs.

File: examples/snr-example.py (hunk 2 above)
Comment (efkag) [reply examples/snr-example.py]: Yes Fit() is useless in this case but I used it there to demonstrate that it can be used safely without breaking anything.
For some of the other metrics this function will actually fir the RFs.
